File: backend/api/schemas/faq.py
from pydantic import BaseModel, Field


class FaqCreate(BaseModel):
    question: str
    answer: str
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class FaqUpdate(BaseModel):
    question: str
    answer: str
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class FaqResponse(BaseModel):
    id: int
    question: str
    answer: str
    position: int
    is_hidden: bool

File: frontend/src/types/faq.ts
export interface FaqCreate {
  question: string
  answer: string
  position?: number
  is_hidden?: boolean
}

export interface FaqUpdate {
  question: string
  answer: string
  position?: number
  is_hidden?: boolean
}

export interface FaqResponse {
  id: number
  question: string
  answer: string
  position: number
  is_hidden: boolean
}